pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -504,3 +518,42 @@ func (ic *ifConfigurator) getInterceptedInterfaces(
 func (ic *ifConfigurator) getOVSInterfaceType() int {
 	return internalOVSInterfaceType
 }
+
+func (ic *ifConfigurator) addPostInterfaceCreateHook(containerID, endpointName string, containerAccess *containerAccessArbitrator, hook postInterfaceCreateHook) error {
+	if containerAccess == nil {
+		return fmt.Errorf("container lock cannot be null")
+	}
+	expectedEP, ok := ic.getEndpoint(endpointName)
+	if !ok {
+		return fmt.Errorf("failed to find HNSEndpoint %s", endpointName)
+	}
+	go func() {
+		ifaceName := fmt.Sprintf("vEthernet (%s)", endpointName)
+		err := wait.PollImmediate(time.Second, 60*time.Second, func() (done bool, err error) {
+			containerAccess.lockContainer(containerID)
+			defer containerAccess.unlockContainer(containerID)
+			currentEP, ok := ic.getEndpoint(endpointName)
+			if !ok {
+				klog.InfoS("HNSEndpoint doesn't exist in cache, exit current goroutine", "HNSEndpoint Name", endpointName)
+				return true, nil
+			}
+			if currentEP.Id != expectedEP.Id {
+				klog.InfoS("Detected HNSEndpoint change, exit current goroutine", "HNSEndpoint Name", endpointName)
+				return true, nil
+			}
+			if !util.HostInterfaceExists(ifaceName) {
+				klog.InfoS("Waiting for interface to be created", "interface name", ifaceName)
+				return false, nil
+			}
+			if err := hook(); err != nil {
+				return false, err
+			}
+			return true, nil
+		})
+
+		if err != nil {
+			klog.Errorf("Failed to configure host interface %s: %v", ifaceName, err)
```

Comment:
```suggestion
			klog.ErrorS(err, "Failed to execute postInterfaceCreateHook", "interface name", ifaceName)
```
And I suggest to declare err and pollErr separately and log the error from the hook instead of the pollErr, otherwise we will loss the actual error here (pollErr will be TimeoutError).
```
var err error
pollErr := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
        ...
        if err = hook(); err != nil {
                return false, err
        }
})

if pollErr != nil {
        klog.ErrorS(err, "Failed to execute postInterfaceCreateHook", "interface name", ifaceName)
}
```